request-cert/main.go

```diff
@@ -52,53 +51,103 @@ func main() {
 
 	// Check certificate type.
 	var template *x509.CertificateRequest
-	var filePrefix string
+	var filename, csrName string
 	var wantServerAuth bool
 
+	hostname, err := os.Hostname()
```

Comment:
For now, I think it's not too terrible to have to start with 1 replica then scale it up to the desired number.
But I agree with you, this needs to be addressed. We'll have a similar problem if stateful sets ever allow parallel pod setup. However, much like persisting the certificates in secrets was deserving of its own PR, I think this does too.